builtin/credential/approle/path_role.go

```diff
@@ -1079,35 +1024,33 @@ func (b *backend) pathRoleRead(ctx context.Context, req *logical.Request, data *
 	}
 
 	respData := map[string]interface{}{
-		"bind_secret_id": role.BindSecretID,
-		// TODO - remove this deprecated field in future versions,
-		// and its associated warning below.
-		"bound_cidr_list":       role.SecretIDBoundCIDRs,
+		"bind_secret_id":        role.BindSecretID,
 		"secret_id_bound_cidrs": role.SecretIDBoundCIDRs,
-		"token_bound_cidrs":     role.TokenBoundCIDRs,
-		"period":                role.Period / time.Second,
-		"policies":              role.Policies,
 		"secret_id_num_uses":    role.SecretIDNumUses,
 		"secret_id_ttl":         role.SecretIDTTL / time.Second,
-		"token_max_ttl":         role.TokenMaxTTL / time.Second,
-		"token_num_uses":        role.TokenNumUses,
-		"token_ttl":             role.TokenTTL / time.Second,
 		"local_secret_ids":      false,
-		"token_type":            role.TokenType,
 	}
+	role.PopulateTokenData(respData)
 
 	if role.SecretIDPrefix == secretIDLocalPrefix {
 		respData["local_secret_ids"] = true
 	}
 
+	// Backwards compat data
+	if role.Period != 0 {
```

Comment:
Oh, is this the reason for retaining the older fields?

Reply:
Yes